test/StringClient.test.js

```diff
@@ -64,4 +61,35 @@ describe('StringClient', function () {
             return done();
         });
     });
+
+
+    it('should honor requestTimeout when socket has already been established',
+    function (done) {
+        SERVER.get('/foo', function (req, res, next) {
+            res.send('foo');
+            return next();
+        });
+
+        SERVER.get('/fooSlow', function (req, res, next) {
+            setTimeout(function () {
+                res.send('foo');
+                return next();
+            }, 200);
+        });
+
+        // first request should establish keep alive
+        CLIENT.get('/foo', function (err, req, res, data) {
+            assert.ifError(err);
+            assert.strictEqual(data, 'foo');
+            // second request should reuse existing socket
```

Comment:
Do we know this? Is there a way to force this a bit more explicitly?

If this behavior changes in the future (i.e. a connection pool), is it possible that this test will continue to pass even if the underlying implementation is broken?

Reply:
restify-clients uses keep alive by default, so this should always be the case. A connection pool wouldn't change this in a meaningful way, since the two choices are either create a new connection or reuse an existing connection. Ironically, I only discovered this issue when I was testing using a connection pool.